app/lib/account/agent.dart

```diff
@@ -81,9 +87,14 @@ abstract class AuthenticatedAgent {
 }
 
 /// Holds the authenticated Github Action information.
+///
+/// The [agentId] has the following format: `service:github-actions:organization/repository`
 class AuthenticatedGithubAction implements AuthenticatedAgent {
   @override
-  String get agentId => KnownAgents.githubActions;
+  late final agentId = [
+    KnownAgents.githubActions,
+    if (payload.hasValidRepository) payload.repository,
```

Comment:
There should always be a `repository` property, right?

Wouldn't it be better to throw an `ArgumentError` in the constructor instead.
This shouldn't happen, if it does aren't we fine with a 500 error.

We could perhaps check when creating a `AuthenticatedGithubAction` that a repository property is present. And refuse to make a `AuthenticatedGithubAction` instance, if not.

If we ever need to support actions that don't have a repository associated, then we can make a `AuthenticatedGithubAction` and a `AuthenticatedGithubRepositoryAction extends AuthenticatedGithubAction`. But we probably don't need both today.

In either case, producing the wrong value seems like a worse option, than return 500.
If it's because we don't know if the repository property is always there, then I think we should test whether it is. I suspect it is always there:
https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect